spec/yoti/dynamic_share_service/policy/wanted_attribute_spec.rb

```diff
@@ -41,6 +41,29 @@
       end
     end
 
+    describe '.with_constraint' do
+      let :constraint do
+        Yoti::DynamicSharingService::SourceConstraint
+          .builder
+          .with_passport
+          .build
+      end
+      let :attribute do
+        Yoti::DynamicSharingService::WantedAttribute
+          .builder
+          .with_constraint(constraint)
+          .build
+      end
+
+      it 'has a source constraint' do
+        expect(attribute.constraints.length).to eql 1
+      end
+
+      it 'marshals to json' do
+        expect(attribute.to_json).to eql '{"name":null,"constraints":[{"type":"SOURCE","preferred_sources":{"anchors":[{"name":"PASSPORT"}]}}]}'
```

Comment:
`"name":null` results in a `400` response - perhaps we should add some validation to ensure a name is provided using `.with_name('some name')` - I'll check that other SDKs validate this